File: mybatis-plus-join-test/test-join/src/main/java/com/github/yulichang/test/join/entity/ID.java
package com.github.yulichang.test.join.entity;

import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.extension.activerecord.Model;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;

public class ID<T extends Serializable> extends Model<ID<T>> {

    @Getter
    @Setter
    @TableId
    private T id;
}

File: mybatis-plus-join-test/test-kotlin/src/main/kotlin/com/github/yulichang/test/kt/entity/ID.kt
package com.github.yulichang.test.kt.entity

import com.baomidou.mybatisplus.annotation.TableId
import com.baomidou.mybatisplus.extension.activerecord.Model
import lombok.Getter
import lombok.Setter
import java.io.Serializable

open class ID<T : Serializable?> : Model<ID<T>?>() {
    @Getter
    @Setter
    @TableId
    var id: T? = null
}
